### ISSUE TITLE
Optimizer.step() & scheduler.step() in wrong order when restoring

### ISSUE THREAD

**Issue by polleyr** [dingmyu/D4LCN#25]
Hello,

first of all, thank you for your amazing work.

I am using pytorch 1.5.1 and I receive the warining that `lr_scheduler.step()` is called before` optimizer.step()` when I restore a checkpoint in order to continue a training. I don't get this warning when I start a new training by scratch.

Is this normal or should I be worried?

Thanks

**Comment by polleyr** [comment]
The `_LRScheduler` class steps the scheduler when getting initialized. I removed this. I also moved the   `adjust_lr` call to the end of the iteration for loop.
